WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m

```diff
@@ -724,47 +724,73 @@ - (UITableViewCell *)configurePostFormatCellForIndexPath:(NSIndexPath *)indexPat
-
-        cell = activityCell;
+        return [self cellForSetFeaturedImage];
 
     } else if (self.isUploadingMedia || self.apost.featuredImage.remoteStatus == MediaRemoteStatusPushing) {
         if (!self.isUploadingMedia) {
             self.isUploadingMedia = YES;
-            [self setupObservingOfMedia: self.apost.featuredImage];
+            [self setupObservingOfMedia:self.apost.featuredImage];
         }
         self.featuredImage = nil;
-        self.progressCell = [self.tableView dequeueReusableCellWithIdentifier:TableViewProgressCellIdentifier forIndexPath:indexPath];
-        [WPStyleGuide configureTableViewCell:self.progressCell];        
-        [self.progressCell setProgress:self.featuredImageProgress];
-        self.progressCell.tag = PostSettingsRowFeaturedLoading;
-        cell = self.progressCell;
+        return [self cellForFeaturedImageUploadProgressAtIndexPath:indexPath];
+
     } else if (self.apost.featuredImage && self.apost.featuredImage.remoteStatus == MediaRemoteStatusFailed) {
-        WPTableViewActivityCell *activityCell = [self getWPTableViewActivityCell];
-        activityCell.textLabel.text = NSLocalizedString(@"Upload failed. Tap for options.", @"Description to show on post setting for a featured image that failed to upload.");
-        activityCell.tag = PostSettingsRowFeaturedImageRemove;
-        cell = activityCell;
-    } else {
-        PostFeaturedImageCell *featuredImageCell = [self.tableView dequeueReusableCellWithIdentifier:TableViewFeaturedImageCellIdentifier forIndexPath:indexPath];
-        featuredImageCell.delegate = self;
-        [WPStyleGuide configureTableViewCell:featuredImageCell];
+        return [self cellForFeaturedImageError];
 
-        NSURL *featuredURL = [NSURL URLWithString:self.apost.featuredImage.remoteURL];
+    } else {
+        NSURL *featuredURL = [self urlForFeaturedImage];
         if (!featuredURL) {
-            featuredURL = self.apost.featuredImageURLForDisplay;
+            return [self cellForFeaturedImageError];
         }
-        [featuredImageCell setImageWithURL:featuredURL inPost:self.apost withSize:self.featuredImageSize];
 
-        cell = featuredImageCell;
-        cell.tag = PostSettingsRowFeaturedImage;
+        return [self cellForFeaturedImageWithURL:featuredURL atIndexPath:indexPath];
     }
+}
 
-    return cell;
+- (UITableViewCell *)cellForSetFeaturedImage
+{
+    WPTableViewActivityCell *activityCell = [self getWPTableViewActivityCell];
+    activityCell.textLabel.text = NSLocalizedString(@"Set Featured Image", @"");
+    activityCell.tag = PostSettingsRowFeaturedImageAdd;
+
+    return activityCell;
+}
+
+- (UITableViewCell *)cellForFeaturedImageError
+{
+    WPTableViewActivityCell *activityCell = [self getWPTableViewActivityCell];
+    activityCell.textLabel.text = NSLocalizedString(@"Upload failed. Tap for options.", @"Description to show on post setting for a featured image that failed to upload.");
```

Comment:
If I understood this right, we don't know why this is happening, so it might not be a failed upload. I think it'd be safer to pretend there's no featured image and show the button to set one instead.

Reply:
I agree.
Just pushed an update with this change :)